source/MaterialXRuntime/RtStage.cpp

```diff
@@ -141,11 +141,30 @@ void RtStage::restorePrim(const RtPath& parentPath, const RtPrim& prim)
     _cast(_ptr)->restorePrim(*static_cast<PvtPath*>(parentPath._ptr), prim);
 }
 
+RtPrim RtStage::getImplementation(const RtNodeDef& definition) const
+{
+    const RtToken& nodeDefName = definition.getName();
+    const RtToken& nodeDefVersion = definition.getVersion();
+
+    RtSchemaPredicate<RtNodeGraph> filter;
+    for (RtPrim child : _cast(_ptr)->getRootPrim()->getChildren(filter))
+    {
+        RtNodeGraph nodeGraph(child);
+        if (nodeGraph.getDefinition() == nodeDefName &&
```

Comment:
Here if the nodedef names match a version is already implied, by the version set on the nodedef. So no extra version control should be needed here, as far as I know.

Reply:
I think that is true. The node->nodedef association needs this.